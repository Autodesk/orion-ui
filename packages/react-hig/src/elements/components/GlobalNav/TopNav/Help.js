/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */
import HIGElement from '../../../HIGElement';
import * as PropTypes from 'prop-types';
import createComponent from '../../../../adapters/createComponent';

export class Help extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      title: 'setTitle',
      link: 'setLink'
    };

    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      if (mapping[propKey]) {
        this.hig[mapping[propKey]](propValue);
      } else {
        this.commitPropChange(propKey, propValue);
      }
    }
  }
}

const HelpComponent = createComponent(Help);

HelpComponent.propTypes = {
  title: PropTypes.string,
  link: PropTypes.string
};

HelpComponent.__docgenInfo = {
  props: {
    title: {
      description: 'sets the title of a Help shortcut'
    },

    link: {
      description: 'sets the link of a Help shortcut'
    }
  }
};

export default HelpComponent;
