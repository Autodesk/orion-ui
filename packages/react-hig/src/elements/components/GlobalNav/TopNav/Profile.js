/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../../HIGElement';
import createComponent from '../../../../adapters/createComponent';

export class Profile extends HIGElement {
  componentDidMount() {
    if (this.initialProps.open === true) {
      this.hig.open();
    }
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    const mapping = {
      image: 'setImage',
      name: 'setName',
      email: 'setEmail',
      profileSettingsLabel: 'setProfileSettingsLabel',
      signOutLabel: 'setSignOutLabel',
      profileSettingsLink: 'setProfileSettingsLink'
    };

    const openIndex = updatePayload.indexOf('open');
    if (openIndex >= 0) {
      const [openKey, openSetting] = updatePayload.splice(openIndex, 2);
      if (openKey) {
        if (openSetting === true) {
          this.hig.open();
        } else {
          this.hig.close();
        }
      }
    }

    this.commitUpdateWithMapping(updatePayload, mapping);
  }
}

const ProfileComponent = createComponent(Profile);

ProfileComponent.propTypes = {
  image: PropTypes.string,
  name: PropTypes.string,
  email: PropTypes.string,
  signOutLabel: PropTypes.string,
  profileSettingsLabel: PropTypes.string,
  profileSettingsLink: PropTypes.string,
  onSignOutClick: PropTypes.func,
  onProfileImageClick: PropTypes.func,
  onProfileClickOutside: PropTypes.func
};

ProfileComponent.__docgenInfo = {
  props: {
    open: {
      description: '{bool} sets whether flyout is open or closed'
    },
    name: {
      description: 'sets {String} display name'
    },
    email: {
      description: 'sets {String} email'
    },
    image: {
      description: 'sets {String} image - url to svg or other image object'
    },
    signOutLabel: {
      description: 'sets {String} label for sign out button'
    },
    profileSettingsLabel: {
      description: 'sets {String} label for settings button'
    },
    profileSettingsLink: {
      description: 'sets {String} url to settings page'
    },
    onSignOutClick: {
      description: '{function} Triggered when user clicks sign out button'
    },
    onProfileImageClick: {
      description: '{function} triggers when you click the profile'
    },
    onProfileClickOutside: {
      description: '{function} triggers when you click anywhere but the profile flyout'
    }
  }
};

export default ProfileComponent;
