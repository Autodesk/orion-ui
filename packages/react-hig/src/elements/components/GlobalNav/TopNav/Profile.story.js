/**
 Copyright 2016 Autodesk,Inc.

 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at

 http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing, software
 distributed under the License is distributed on an "AS IS" BASIS,
 WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.

 */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
import GlobalNav from '../GlobalNav';

import logo from '../../../../images/bim-logo.png';
import profileImage from '../../../../images/profileImage.png';

const TopNav = GlobalNav.TopNav;
const Profile = TopNav.Profile;
const Slot = GlobalNav.Slot;

function ParagraphSlot() {
  return (
    <Slot>
      <p>
        Next level deep v roof party, jianbing pok pok pug butcher vape farm-to-table kombucha. Yr snackwave VHS, wolf
        poutine actually woke poke flexitarian paleo food truck DIY kale chips viral yuccie. Cornhole tattooed vaporware
        affogato, gentrify mlkshk portland organic. Swag try-hard cronut hashtag, etsy bespoke chia banjo messenger bag.
        Mustache umami godard man braid cronut yuccie. YOLO vaporware franzen, gochujang typewriter mixtape brunch salvia
        paleo lyft. Four dollar toast tumblr mustache thundercats single-origin coffee, freegan flexitarian cold-pressed
        beard roof party VHS venmo af ugh bushwick.
      </p>
    </Slot>
  );
}

storiesOf('TopNav Profile', module)
  .addWithInfo(
    'default',
    <div>
      Basic usage of <pre>GlobalNav.TopNav.Profile</pre>

    </div>,
    () => (
      <GlobalNav>
        <TopNav logo={logo}>
          <Profile
            image={profileImage}
            name="Jane Designer"
            email="jane.designer@example.com"
            onProfileSettingsClick={action('profile settings clicked')}
            onSignOutClick={action('sign out clicked')}
          />
        </TopNav>
        <ParagraphSlot />
      </GlobalNav>
    )
  )
  .addWithInfo(
    'different initialSettings (no knobs)',
    <div>
      TopNav Search has different initial settings (knobs trigger a commitUpdate)
    </div>,
    () => (
      <GlobalNav>
        <TopNav logo={logo}>
          <Profile
            image={profileImage}
            email="jane.designer@example.com"
            name="Jane Designer"
            open={true}
            signOutLabel="Sign off"
            signOutLink="http://www.autodesk.com"
            onProfileSettingsClick={function() {
              alert('profile settings clicked');
            }}
          />
        </TopNav>
        <ParagraphSlot />
      </GlobalNav>
    )
  )
  .addWithInfo('user configured', `Set open state, events, `, () => {
    return (
      <GlobalNav>
        <TopNav logo={logo}>
          <Profile
            image={text('image', profileImage)}
            name={text('name', 'Jane Designer')}
            email={text('email', 'jane.designer@example.com')}
            open={boolean('profileFlyoutOpen', false)}
            onProfileClickOutside={action('Clicked outside profile')}
            onProfileImageClick={action('profile image clicked')}
            onProfileSettingsClick={action('profile settings clicked')}
            profileSettingsLabel={text('profileSettingsLabel', 'My Account')}
            profileSettingsLink={text('profileSettingsLink', '#')}
            onSignOutClick={action('sign out clicked')}
            signOutLabel={text('signOutLabel', 'Logout')}
            signOutLink={text('signOutLink', '/signout')}
          />
        </TopNav>
        <ParagraphSlot />
      </GlobalNav>
    );
  });
