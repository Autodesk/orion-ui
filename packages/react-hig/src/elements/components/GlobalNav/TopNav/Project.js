/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../../HIGElement';
import createComponent from '../../../../adapters/createComponent';

export class Project extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.props = { ...initialProps };
    this.callOnActivateCallback = this.callOnActivateCallback.bind(this);
  }

  componentDidMount() {
    this.hig.onClick(this.callOnActivateCallback);
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.props = { ...this.props, ...updatePayload };

    this.processUpdateProps(updatePayload)
      .mapToHIGFunctions({
        image: 'setImage',
        label: 'setLabel'
      })
      .mapToHIGEventListeners(['onClick'])
      .handle('active', value => {
        if (value) {
          this.hig.activate();
          this.onActivate(this);
        } else {
          this.hig.deactivate();
        }
      });
  }

  onActivate(callback) {
    this._onActivate = callback;
  }

  callOnActivateCallback() {
    if (!this._onActivate) {
      return
    }

    this._onActivate(this);
  }
}

const ProjectComponent = createComponent(Project);

ProjectComponent.propTypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  onClick: PropTypes.func
};

ProjectComponent.__docgenInfo = {
  props: {
    label: {
      description: 'sets {String} the label displayed for a project in project/account switcher'
    },
    image: {
      description: 'sets {String} the image displayed for a project in project/account switcher'
    },
    activate: {
      description: '{func} activates the project'
    },
    deactivate: {
      description: '{func} deactivates the project'
    },
    onClick: {
      description: '{func} calls the provided callback when user clicks on the project'
    }
  }
};

export default ProjectComponent;
