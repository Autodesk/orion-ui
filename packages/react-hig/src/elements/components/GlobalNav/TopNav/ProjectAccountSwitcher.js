/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../../HIGElement';
import HIGNodeList from '../../../HIGNodeList';
import HIGChildValidator from '../../../HIGChildValidator';
import createComponent from '../../../../adapters/createComponent';
import AccountComponent, { Account } from './Account';
import ProjectComponent, { Project } from './Project';

export class ProjectAccountSwitcher extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.accounts = new HIGNodeList({
      type: Account,
      HIGConstructor: this.hig.partials.Account,
      onAdd: (instance, beforeInstance) => {
        this.hig.addAccount(instance, beforeInstance);
      }
    });

    this.projects = new HIGNodeList({
      type: Project,
      HIGConstructor: this.hig.partials.Project,
      onAdd: (instance, beforeInstance) => {
        this.hig.addProject(instance, beforeInstance);
      }
    });

    ['_render', 'openFlyout', 'closeFlyout'].forEach(fn => {
      this[fn] = this[fn].bind(this);
    });

    this.props = { ...initialProps };
    this.state = {};
  }

  componentDidMount() {
    this.accounts.componentDidMount();
    this.projects.componentDidMount();

    this.commitUpdate(this.props);
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.processUpdateProps(updatePayload)
      .mapToHIGFunctions({
        activeImage: 'setActiveImage',
        activeLabel: 'setActiveLabel',
        activeType: 'setActiveType'
      })
      .mapToHIGEventListeners(['onClick', 'onClickOutside'])
      .handle('open', value => {
        this.props.open = value;
      })
      .then(this._render);
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case Project:
        return this.projects.createElement(ElementConstructor, props);
      case Account:
        return this.accounts.createElement(ElementConstructor, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  insertBefore(instance, beforeChild = {}) {
    if (instance instanceof Account) {
      this.accounts.insertBefore(instance);
    } else if (instance instanceof Project) {
      this.projects.insertBefore(instance);
    } else {
      throw new Error(`${this.constructor.name} cannot have a child of type ${instance.constructor.name}`);
    }
  }

  openFlyout() {
    this.state.open = true;
    this._render();
  }

  closeFlyout() {
    this.state.open = false;
    this._render();
  }

  _render() {
    let open = this.props.open;
    if (open === undefined) {
      open = this.state.open;
    }
    open ? this.hig.open() : this.hig.close();

    if (this.projects.length > 1 || this.accounts.length > 1) {
      this.hig.addCaret();
      this.configureHIGEventListener('onClick', this.openFlyout);
      this.configureHIGEventListener('onClickOutside', this.closeFlyout);
    } else {
      this.hig.removeCaret();
      this.configureHIGEventListener('onClick', undefined);
      this.configureHIGEventListener('onClickOutside', undefined);
    }
  }
}

const ProjectAccountSwitcherComponent = createComponent(ProjectAccountSwitcher);

ProjectAccountSwitcherComponent.propTypes = {
  open: PropTypes.bool,
  activeLabel: PropTypes.string,
  activeImage: PropTypes.string,
  activeType: PropTypes.string,
  onClick: PropTypes.func,
  onClickOutside: PropTypes.func,
  children: HIGChildValidator([AccountComponent, ProjectComponent])
};

ProjectAccountSwitcherComponent.__docgenInfo = {
  props: {
    open: {
      description: '{bool} opens the project/account switcher'
    },
    activeLabel: {
      description: 'sets {String} the label displayed in the top nav'
    },
    activeImage: {
      description: 'sets {String} the image displayed in the top nav'
    },
    activeType: {
      description: 'sets the {String} type of the item displayed in the top nav'
    },
    onClick: {
      description: '{func} calls the provided callback when user clicks on the switcher in the top nav'
    },
    onClickOutside: {
      description: '{func} calls the provided callback when the switcher is open and the user clicks outside the switcher'
    },
    children: {
      description: 'support adding Project and Account'
    }
  }
};

ProjectAccountSwitcherComponent.Account = AccountComponent;
ProjectAccountSwitcherComponent.Project = ProjectComponent;

export default ProjectAccountSwitcherComponent;
