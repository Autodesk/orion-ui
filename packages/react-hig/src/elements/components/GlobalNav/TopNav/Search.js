/* Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */

import HIGElement from '../../../HIGElement';
import * as PropTypes from 'prop-types';
import createComponent from '../../../../adapters/createComponent';

export class Search extends HIGElement {
  componentDidMount() {
    this.hig.onInput(this.topNavSearchOnInput.bind(this));
    this.hig.onClearIconClick(this.topNavClearInput.bind(this));
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      placeholder: 'setPlaceholder',
    };
    super.commitUpdateWithMapping(updatePayload, mapping);
  }

  topNavSearchOnInput(event) {
    this.hig.showClearIcon();
    this.hig.setQuery(event.target.value);
  };

  topNavClearInput(event) {
    this.hig.setQuery("");
    this.hig.hideClearIcon();
  };

}

const SearchComponent = createComponent(Search);

SearchComponent.propTypes = {
  placeholder: PropTypes.string,
  query: PropTypes.string,
  showClearIcon: PropTypes.func,
  hideClearIcon: PropTypes.func,
  onClearIconClick: PropTypes.func,
  onInput: PropTypes.func,
  onFocusIn: PropTypes.func,
  onFocusOut: PropTypes.func
};

SearchComponent.__docgenInfo = {
  props: {
    setPlaceholder: { description: "{String} sets the placeholder text" },
    setQuery: { description: "{String} sets the search query text" },
    showClearIcon: { description: "{function} shows the clear input icon" },
    hideClearIcon: { description: "{function} hides the clear input icon" },
    onClearIconClick: { description: "{function} Triggers when user clicks clear icon" },
    onInput: { description: "{function} Triggers when user enters text into search field"},
    onFocusIn: { description: "{function} Triggers when input has focus"},
    onFocusOut: { description: "{function} Triggers when input loses focus"}
  }
};
export default SearchComponent;
