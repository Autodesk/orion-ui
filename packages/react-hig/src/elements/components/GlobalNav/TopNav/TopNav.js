/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
//import * as HIG from 'hig.web';
import * as PropTypes from 'prop-types';

import HIGElement from '../../../HIGElement';
import HIGChildValidator from '../../../HIGChildValidator';
import createComponent from '../../../../adapters/createComponent';

import HIGNodeList from '../../../HIGNodeList';
import ProjectAccountSwitcherComponent, {
  ProjectAccountSwitcherAdapter
} from '../../../../adapters/ProjectAccountSwitcherAdapter';
import ProfileComponent, {
  ProfileAdapter
} from '../../../../adapters/ProfileAdapter';
import ShortcutComponent, { Shortcut } from './Shortcut';
import HelpComponent, { Help } from './Help';
import SearchComponent, { Search } from './Search';

export class TopNav extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.shortcuts = new HIGNodeList({
      type: Shortcut,
      HIGConstructor: this.hig.partials.Shortcut,
      onAdd: (instance, beforeInstance) => {
        this.hig.addShortcut(instance, beforeInstance);
      }
    });
  }

  componentDidMount() {
    // Add any children
    if (this.profile) {
      this.hig.addProfile(this.profile.hig);
      this.profile.mount();
    }

    if (this.projectAccountSwitcher) {
      this.hig.addProjectAccountSwitcher(this.projectAccountSwitcher.hig);
      this.projectAccountSwitcher.mount();
    }

    this.shortcuts.componentDidMount();

    if (this.help) {
      this.hig.addHelp(this.help.hig);
      this.help.mount();
    }

    if (this.search) {
      this.hig.addSearch(this.search.hig);
      this.search.mount();
    }
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.processUpdateProps(updatePayload).mapToHIGFunctions({
      logo: 'setLogo',
      logoLink: 'setLogoLink'
    });
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case ProfileAdapter:
        return new ProfileAdapter(this.hig.partials.Profile, props);
      case ProjectAccountSwitcherAdapter:
        return new ProjectAccountSwitcherAdapter(
          this.hig.partials.ProjectAccountSwitcher,
          props
        );
      case Help:
        return new Help(this.hig.partials.Help, props);
      case Shortcut:
        return this.shortcuts.createElement(ElementConstructor, props);
      case Search:
        return new Search(this.hig.partials.Search, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    this.requireSingleInstance(instance);
    this.checkValidChild(instance);

    if (instance instanceof Shortcut) {
      this.shortcuts.insertBefore(instance);
    } else {
      this[this.getPropertyNameFor(instance)] = instance;
      if (this.mounted) {
        this.hig[this.getFunctionNameFor(instance)](instance.hig);
        instance.mount();
      }
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof ProfileAdapter) {
      this.profile = null;
    }

    instance.unmount();
  }

  requireSingleInstance(instance) {
    const requiredSingle = [
      'Profile',
      'ProjectAccountSwitcher',
      'Help',
      'Search'
    ];
    super.requireSingleInstance(instance, requiredSingle);
  }

  checkValidChild(instance) {
    const validInstances = [
      ProfileAdapter,
      ProjectAccountSwitcherAdapter,
      Shortcut,
      Help,
      Search
    ];
    if (!validInstances.includes(instance.constructor)) {
      throw new Error(
        `${instance.constructor.name} is not a valid child element of this parent.`
      );
    }
  }

  getPropertyNameFor(instance) {
    if (instance instanceof ProfileAdapter) {
      return 'profile';
    }
    if (instance instanceof ProjectAccountSwitcherAdapter) {
      return 'projectAccountSwitcher';
    }
    if (instance instanceof Shortcut) {
      return 'shortcut';
    }
    if (instance instanceof Help) {
      return 'help';
    }
    if (instance instanceof Search) {
      return 'search';
    }
    return null;
  }

  getFunctionNameFor(instance) {
    return 'add' + instance.constructor.name;
  }

  onHamburgerClick(callback) {
    this.hig.onHamburgerClick(callback);
  }
}

const TopNavComponent = createComponent(TopNav);

TopNavComponent.propTypes = {
  logo: PropTypes.string,
  logoLink: PropTypes.string,
  onHamburgerClick: PropTypes.func,
  addProfile: PropTypes.func,
  addProjectAccountSwitcher: PropTypes.func,
  addSearch: PropTypes.func,
  children: HIGChildValidator([
    ProfileComponent,
    ProjectAccountSwitcherComponent,
    ShortcutComponent,
    HelpComponent,
    SearchComponent
  ])
};

TopNavComponent.__docgenInfo = {
  props: {
    logo: {
      description: 'sets the logo'
    },

    logoLink: {
      description: 'sets the link of the logo'
    },

    onHamburgerClick: {
      description: 'triggered when hamburger icon is clicked'
    },

    addProfile: {
      description: 'adds Profile to the top nav'
    },

    addProjectAccountSwitcher: {
      description: 'Pass in an instance of a ProjectAccountSwitcher partial to mount it to the TopNav'
    },

    addSearch: {
      description: 'Pass in an instance of a topNavSearch partial to mount to TopNav'
    }
  }
};

TopNavComponent.Profile = ProfileComponent;
TopNavComponent.Shortcut = ShortcutComponent;
TopNavComponent.Help = HelpComponent;
TopNavComponent.ProjectAccountSwitcher = ProjectAccountSwitcherComponent;
TopNavComponent.Search = SearchComponent;

export default TopNavComponent;
