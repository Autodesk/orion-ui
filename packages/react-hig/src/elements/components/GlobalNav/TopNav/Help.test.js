/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */
import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from '../../../../adapters/GlobalNav/GlobalNavAdapter';
import TopNav from './TopNav';
import Help from './Help';

const Context = props => {
  return (
    <GlobalNav>
      <TopNav>
        <Help title={props.title} link={props.link} />
      </TopNav>
    </GlobalNav>
  );
};

function setupReactContext(props) {
  const reactContainer = document.createElement('div');
  mount(<Context {...props} />, { attachTo: reactContainer });
  return { reactContainer };
}

describe('Help', () => {
  describe('constructor', () => {
    it('has a good snapshot', () => {
      const props = { title: 'HELP', link: '/help' };
      const { reactContainer } = setupReactContext(props);
      expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
      var elems = reactContainer.getElementsByClassName(
        'hig__global-nav__top-nav__shortcut__link__icon--help'
      );
      expect(elems.length).toEqual(1);
    });
  });
});
