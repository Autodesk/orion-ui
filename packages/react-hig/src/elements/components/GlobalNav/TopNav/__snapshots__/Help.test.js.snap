// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help constructor has a good snapshot 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
        <div class=\\"hig__global-nav__top-nav__hamburger\\">
            <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"30\\" height=\\"30\\" viewBox=\\"0 0 30 30\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-92 -185)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect x=\\"92\\" y=\\"185\\" width=\\"30\\" height=\\"30\\"></rect><path d=\\"M100,195 L115,195 L115,196 L100,196 L100,195 Z M100,200 L115,200 L115,201 L100,201 L100,200 Z M100,205 L115,205 L115,206 L100,206 L100,205 Z\\" fill=\\"#34495E\\"></path></g></svg></div></div>
            <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><title>hig_icon_xclose</title><g fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect transform=\\"rotate(45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect transform=\\"rotate(-45 12 12)\\" x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
        </div>
        <div class=\\"hig__global-nav__top-nav__logo\\">
            <a href=\\"#\\">
                <img src=\\"link\\">
            </a>
        </div>
    </div>
    <!--SEARCH-->
    <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
    <!--PROJECT_ACCOUNT_SWITCHER-->
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
        <div class=\\"hig__global-nav__top-nav__shortcut\\">
  <a href=\\"/help\\" title=\\"HELP\\" class=\\"hig__global-nav__top-nav__shortcut__link\\">
    <div class=\\"hig__global-nav__top-nav__shortcut__link__icon hig__global-nav__top-nav__shortcut__link__icon--help\\">
      <div class=\\"hig__icon\\"><svg viewBox=\\"0.0 0.0 74 74\\" fill=\\"none\\" stroke-linecap=\\"square\\" stroke-miterlimit=\\"10\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"help\\"><clippath id=\\"a\\"><path d=\\"m0 0l100.0 0l0 100.0l-100.0 0l0 -100.0z\\"></path></clippath><g clip-path=\\"url(#a)\\"><path fill=\\"#000\\" fill-opacity=\\"0\\" d=\\"m0 0l100.0 0l0 100.0l-100.0 0z\\" fill-rule=\\"evenodd\\"></path><path fill=\\"#000\\" fill-opacity=\\"0\\" d=\\"m2.2283463 35.761154l0 0c0 -18.61243 15.088358 -33.700787 33.700787 -33.700787l0 0c8.938004 0 17.50993 3.5506084 23.830055 9.870732c6.3201256 6.3201227 9.870731 14.892051 9.870731 23.830055l0 0c0 18.61243 -15.088356 33.70079 -33.700787 33.70079l0 0c-18.61243 0 -33.700787 -15.08836 -33.700787 -33.70079z\\" fill-rule=\\"evenodd\\"></path><path stroke=\\"#000\\" stroke-width=\\"3\\" stroke-linejoin=\\"round\\" stroke-linecap=\\"butt\\" d=\\"m2.2283463 35.761154l0 0c0 -18.61243 15.088358 -33.700787 33.700787 -33.700787l0 0c8.938004 0 17.50993 3.5506084 23.830055 9.870732c6.3201256 6.3201227 9.870731 14.892051 9.870731 23.830055l0 0c0 18.61243 -15.088356 33.70079 -33.700787 33.70079l0 0c-18.61243 0 -33.700787 -15.08836 -33.700787 -33.70079z\\" fill-rule=\\"evenodd\\"></path><path fill=\\"#000\\" d=\\"m38.878353 43.981777l-5.984375 0q-0.015625 -1.296875 -0.015625 -1.578125q0 -2.90625 0.953125 -4.78125q0.96875 -1.875 3.84375 -4.21875q2.890625 -2.34375 3.453125 -3.0625q0.859375 -1.15625 0.859375 -2.53125q0 -1.921875 -1.53125 -3.296875q-1.53125 -1.375 -4.140625 -1.375q-2.5 0 -4.1875 1.4375q-1.6875 1.421875 -2.328125 4.359375l-6.046875 -0.75q0.265625 -4.203125 3.578125 -7.125q3.3125 -2.9375 8.703125 -2.9375q5.671875 0 9.015625 2.96875q3.359375 2.96875 3.359375 6.90625q0 2.171875 -1.234375 4.125q-1.21875 1.9375 -5.25 5.296875q-2.09375 1.734375 -2.59375 2.796875q-0.5 1.046875 -0.453125 3.765625zm-5.984375 8.859375l0 -6.59375l6.59375 0l0 6.59375l-6.59375 0z\\"></path><path fill=\\"#000\\" fill-opacity=\\"0\\" d=\\"m69.62992 67.40682l426.6772 0l0 49.76378l-426.6772 0z\\" fill-rule=\\"evenodd\\"></path></g></svg></div>
    </div>
  </a>
</div><!--SHORTCUT-->
    </div>
    <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
